src/citrine/resources/data_concepts.py

```diff
@@ -496,15 +518,17 @@ def poll_async_update_job(self, job_id: UUID, *, timeout: float = 2 * 60,
         # That worked, nothing returned in this case
         return None
 
-    def get(self, uid: Union[UUID, str], scope: str = CITRINE_SCOPE) -> ResourceType:
+    def get(self, uid: Union[UUID, str, LinkByUID, BaseEntity],
```

Comment:
We should probably be implementing this as a custom type so that we don't have this complex pattern all over the place.

Reply:
I'm not sure I agree. It would be more compact, but the custom type would be an artifice that only serves to save space. I don't think it's bad to be explicit when saying "we accept any of these 4 types which are _not_ the same but carry equivalent information"

Reply:
Compactness carries with it resilience to maintenance bugs and ease of reading.  Given how we have a substantial number of methods that accept `UUID, str, LinkByUID` plus a subclass of `BaseEntity`, there's value is having the signatures read `Union[IdThing, BaseEntity]`, so you know what to expect about the variability.